scripts/remote-provision.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+
+set -o errexit   # abort on nonzero exit status
+set -o errtrace  # pass ERR trap down to functions, substitutions, etc
+set -o nounset   # abort on unbound variable
+set -o pipefail  # don’t hide errors within pipes
+
+[[ "${BASH_VERSINFO:-0}" -ge 4 ]] || { echo "Bash version 4 or higher is required."; exit 1; }
+
+usage() {
+    echo "Usage $0 [OPTIONS] <teleport-tunnel|ipv4>
+
+    Options:
+        -r, --reprovision   Re-provisioning the device and skips fetching of attestation certificate
+        -p, --passphrase    Expected to be used with IPv4 (required for dev images with ssh access)
+        -t, --plug-trust    Path for plug_and_trust.tar.gz on the host (required for prod images).
+                            Check the worldcoin/plug-and-trust/releases for the appropriate version
+                            based on the device's software release
+    Example:
+        $0 -r -t ~/Downloads/plug_and_trust.tar.gz <teleport-tunnel>"
+}
+
+provision_device() {
+    local remote="${1}"
+    local short="${2}"
+    local ssh_prefix="${3}"
+    local plug_trust="${4}"
+
+    local user="worldcoin"
+    local se_dir="/usr/persistent/se"
+    local key_dir="${se_dir}/keystore"
+    if [[ "${ssh_prefix}" == "tsh" ]]; then
+        user="root"
+    fi
+
+    # If /se/keystore is not present, provisioning process was never executed, or the folder was deleted by the user
+    # In this case, reprovisioniong (--short flag) is not allowed for fear of wiping the attestation certificate
+    if [[ ${short} ]]; then
+        ${ssh_prefix} ssh "${user}@${remote}" << EOF
+        set -euo
+        sudo su || true
+        [[ -d ${key_dir} ]] || echo "${key_dir} not exist, re-provisioning blocked" && exit 1
+        mount -o remount,exec /tmp
+        systemctl stop nv-tee-supplicant.service
+        cp /${key_dir}/f0000013.cert /usr/persistent/ || true
+        rm -rf /usr/persistent/tee/ ${key_dir}
+        systemctl start nv-tee-supplicant.service
+        su worldcoin -c "cd -- ${se_dir}; \
+            /${plug_trust}/delete-all.sh || true; /${plug_trust}/provision.sh --short"
+        cp /usr/persistent/f0000013.cert ${key_dir} || true
+        exit 0
+EOF
+    else
+        ${ssh_prefix} ssh "worldcoin@${remote}" << EOF
+        set -euo
+        mkdir -p ${se_dir}
+        cd -- ${se_dir}
+        /service_mode/provision.sh
+EOF
+    fi
+}
+
+main() {
+    local arg
+    local remote
+    local plug_trust=""
+    local short=false
+    local passphrase=""
+    local ssh_prefix="tsh"
+    local positional_args=()
+
+    while [[ $# -gt 0 ]]; do
+        arg="${1}"; shift
+        case ${arg} in
+            -h | --help)
+                usage; exit 0 ;;
+            -r | --reprovision)
+                short=true ;;
```

Comment:
why do we call "reprovision" as "short" ? for me it's a bit misleading?

Reply:
I mean I do see why we call it short, but I don't agree with it. I think variable should be called `reprovision` for clarity.

Reply:
to be inline with the arguments in `plug-and-trust` repo. 
Changed to avoid confusion